WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
> In this particular case include Cachable doesn't make sense if it's not in a model inheriting from ActiveRecord::Base

The same thing could be said about most modules in Ruby. E.g. `Comparable` requires you to implement `<=>`, `Enumerable` requires you to implement `each`. If you think about `all` as the required interface, it doesn't sound too bad.

> which is why extending ActiveRecord::Base feels like "the" thing to do for me.

Sharing code through inheritance has its own drawbacks, though. (`include` is basically multiple inheritance.)

The approach I'd prefer in a language like Ruby is through an explicit wrapper: `FindCacher.new(User)`. But that's not very easy to accomplish once the entire code base is sprinkled with class constants and you have no way to control the instantiation of the classes responsible for fetching data. In a pure language I'd build trees of future requests instead of doing the requests on the spot which makes caching/batching very easy (see [`haxl`](https://github.com/facebook/Haxl) for an example).

tl;dr: Whatever works is fine. :)